### PR TITLE
Fixes VSTS Bug 1002571: Regression: VSC checkout message dialogs shown below progress dialog window.

### PR DESCRIPTION
**Pull request by mkrueger**
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1002571

+ Added missing API for AskQuestion - only method which didn't take an optional parent window.

### CODE REVIEW

File: main/src/addins/MacPlatform/Dialogs/MacAlertDialogHandler.cs [hunk 1]
```diff
@@ -240,12 +240,17 @@ public bool Run (AlertDialogData data)
 							NSApplication.SharedApplication.StopModal ();
 						});
 
-						// pass parent and not alert so that the Runloop will change
-						// and processing will stop until the sheet is closed.
-						// If we pass alert, then it will run until a second alert is created
-						// which will be shown as a dialog and then the runloop changes and
-						// processing stops
-						NSApplication.SharedApplication.RunModalForWindow (parent);
+						if (IdeApp.Workbench.RootWindow?.Visible == true) {
```
Comment (Therzok) [inline main/src/addins/MacPlatform/Dialogs/MacAlertDialogHandler.cs]: deduplicate the method call.

```csharp
Window modal = IdeApp.Workbench.RootWindow?.Visible == true ? parent : alert.Window;
NSApplication.SharedApplication.RunModalForWindow (modal);
```

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.ProgressMonitoring/MessageDialogProgressMonitor.cs [hunk 2]
```diff
@@ -39,7 +39,9 @@ public class MessageDialogProgressMonitor: ProgressMonitor
 		ProgressDialog dialog;
 		bool hideWhenDone;
 		bool showDetails;
-		
+
+		public ProgressDialog ProgressDialog { get => dialog; }
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.ProgressMonitoring/MessageDialogProgressMonitor.cs]: Could be simpler:

```suggestion
		public ProgressDialog ProgressDialog => dialog;
```

File: main/src/addins/MacPlatform/Dialogs/MacAlertDialogHandler.cs (hunk 1 above)
Comment (mkrueger) [reply main/src/addins/MacPlatform/Dialogs/MacAlertDialogHandler.cs]: No, I want to state that this is a work around for something. That would be shadowed by that.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.ProgressMonitoring/MessageDialogProgressMonitor.cs (hunk 2 above)
Comment (mkrueger) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.ProgressMonitoring/MessageDialogProgressMonitor.cs]: Is that only a getter ?

**Comment by sevoku** [comment]
@monojenkins backport release-8.4
@monojenkins backport release-8.3

**Pull request merged**